File: newBot/services/poet_service.py
from .backend_client import BackendClient

class PoetService:
    def __init__(self) -> None:
        self.client = BackendClient()

    def register_poet(
        self,
        user_id: int,
        full_name: str,
        city: str,
        inn: str,
        phone: str,
        business_type: str,
        bik: str,
        account: str,
        bank_name: str,
        bank_ks: str,
        bank_details: str,
    ) -> dict:
        payload = {
            "userId": user_id,
            "fullName": full_name,
            "city": city,
            "inn": inn,
            "phone": phone,
            "businessType": business_type,
            "bik": bik,
            "account": account,
            "bankName": bank_name,
            "bankKs": bank_ks,
            "bankDetails": bank_details,
        }
        return self.client.post("poet/bot", payload)

    def approve_poet(self, user_id: int) -> bool:
        self.client.put(f"poet/bot/{user_id}", {"approved": True})
        return True

    def sign_poet_contract(self, user_id: int) -> None:
        self.client.put(f"poet/bot/{user_id}", {"contractSigned": True})

    def get_poet_profile(self, user_id: int) -> dict:
        return self.client.get(f"poet/bot/{user_id}")

    def remove_poet(self, user_id: int) -> bool:
        self.client.delete(f"poet/bot/{user_id}")
        return True

File: backend/src/modules/poet/poet.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from '@/src/core/prisma/prisma.service';
import { plainToInstance } from 'class-transformer';
import { CreatePoetDto } from './dto/create-poet.dto';
import { UpdatePoetDto } from './dto/update-poet.dto';
import { PoetResponseDto } from './dto/poet-response.dto';

@Injectable()
export class PoetService {
    constructor(private readonly prismaService: PrismaService) { }

    async create(data: CreatePoetDto): Promise<PoetResponseDto> {
        const poet = await this.prismaService.poet.create({
            data: {
                user: { connect: { id: data.userId } },
                fullName: data.fullName,
                city: data.city,
                inn: data.inn,
                phone: data.phone,
                businessType: data.businessType,
                bik: data.bik,
                account: data.account,
                bankName: data.bankName,
                bankKs: data.bankKs,
                bankDetails: data.bankDetails,
                approved: data.approved,
                contractSigned: data.contractSigned,
            },
            include: { user: true },
        });
        return plainToInstance(PoetResponseDto, { ...poet.user, ...poet });
    }

    async findAll(): Promise<PoetResponseDto[]> {
        const poets = await this.prismaService.poet.findMany();
        return poets.map((p) => plainToInstance(PoetResponseDto, p));
    }

    async findOne(id: number): Promise<PoetResponseDto> {
        const poet = await this.prismaService.poet.findUnique({ where: { id } });
        if (!poet) {
            throw new NotFoundException('Poet not found');
        }
        return plainToInstance(PoetResponseDto, poet);
    }

    async update(id: number, data: UpdatePoetDto): Promise<PoetResponseDto> {
        const poet = await this.prismaService.poet.update({ where: { userId: id }, data });
        return plainToInstance(PoetResponseDto, poet);
    }

    async remove(id: number): Promise<PoetResponseDto> {
        const poet = await this.prismaService.poet.delete({ where: { id } });
        return plainToInstance(PoetResponseDto, poet);
    }
}